docs/design-documents/design_template.md

```diff
@@ -0,0 +1,194 @@
+# Mbed-OS Design Document
+Write your feature title above, in this case its "Mbed-OS Design Document"
+
+# Table of Contents
+- [Mbed-OS Design Document](#mbed-os-design-document)
+- [Table of Contents](#table-of-contents)
+    - [Revision history](#revision-history)
+- [Introduction](#introduction)
+    - [Overview and Background](#overview-and-background)
+    - [Requirements and Assumptions](#requirements-and-assumptions)
+- [System Architecture and High-Level Design](#system-architecture-and-high-level-design)
+    - [System Architecture and Component Interaction](#system-architecture-and-component-interaction)
+- [Detailed Design](#detailed-design)
+- [Usage Scenarios and Examples](#usage-scenarios-and-examples)
+- [Tools and Configuration Changes](#tools-and-configuration-changes)
+- [Other Information](#other-information)
+    - [Reusability](#reusability)
+    - [Deprecations](#deprecations)
+    - [References](#references)
+    - [Custom Headings Here](#custom-headings-here)
+
+### Revision history
+1.0 - A brief description for this version, e.g. Initial revision - Author name - Date
```

Comment:
this revision history is not needed here (sufficient to have it in git)

Reply:
Please see my comment above.